### ISSUE TITLE
regression: false positive from `clang-analyzer-unix.StdCLibraryFunctions` with `getcwd()`

### ISSUE THREAD

**Issue by seanm** [llvm/llvm-project#128882]
clang-tidy / clang static analyzer tells me:

```
warning: The 1st argument to 'getcwd' is NULL but should not be NULL [clang-analyzer-unix.StdCLibraryFunctions]

         char* currentPath = getcwd(nullptr, 0);
                             ^
```

But the macOS man page for getcwd says:

"If buf is NULL [1st param], space is allocated as necessary to store the pathname and size [2nd param] is ignored.  This space may later be free(3)'d."

Looks like the responsible line is : https://github.com/llvm/llvm-project/blob/defe43bbffb0d25ec468f0e54b20548ec192ff90/clang/lib/StaticAnalyzer/Checkers/StdLibraryFunctionsChecker.cpp#L2662C1-L2662C46

from this commit: https://github.com/llvm/llvm-project/commit/7dd20637c801b429f2dd1040941d00141459d64e

from this PR: https://github.com/llvm/llvm-project/pull/77040

@benshi001

**Comment by FantasqueX** [comment]
i want to work on this, please assign me.

**Comment by seanm** [comment]
@FantasqueX are you going to work on this soon? If not, I will, because it looks pretty trivial to fix...

**Comment by seanm** [comment]
Patch: https://github.com/llvm/llvm-project/pull/135720

@FantasqueX maybe you could review?
